lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -381,15 +383,13 @@ instance NFData WalletDelegation
 
 instance Buildable WalletDelegation where
     build (WalletDelegation act []) =
-        "current wallet delegation: " <> build act <> ", awaiting no change"
+        "delegating to " <> build act
     build (WalletDelegation act [n]) =
-        "current wallet delegation: " <> build act <> ", awaiting " <> build n
+        "delegating to " <> build act <> " → " <> build n
     build (WalletDelegation act [n1, n2]) =
-        "current wallet delegation: " <> build act <> ", awaiting first " <>
-        build n1 <> " then awaiting " <> build n2
+        "delegating to " <> build act <> " → " <> build n1 <> " → " <> build n2
     build (WalletDelegation act _) =
-        "current wallet delegation: " <> build act <>
-        ", something wrong with awaiting"
+        "delegating to " <> build act <> " something wrong with what's next"
```

Comment:
Alternatively:

```hs
instance Buildable WalletDelegation where
    build (WalletDelegation act []) =
        "delegating to " <> build 
    build (WalletDelegation act xs) =
         build (WalletDelegation act []) <> " → " <> build (T.intercalate " → " $ build <$> xs)
```

- :heavy_check_mark: avoid repetition of the literal string
- :heavy_check_mark: handles next uniformly

Reply:
in the end I have this : 
```
385 instance Buildable WalletDelegation where                                                                                                                                                 
386     build (WalletDelegation act []) =                                                                                                                                                     
387         "delegating to " <> build act                                                                                                                                                     
388     build (WalletDelegation act xs) =                                                                                                                                                     
389         build (WalletDelegation act [])                                                                                                                                                   
390         <> " → "                                                                                                                                                                          
391         <> build (T.intercalate " → " $                                                                                                                                                   
392                   (LT.toStrict . Builder.toLazyText . build) <$> xs)
```

Reply:
T.intercalate accepts [Text] as second argument, not [Builder]

Reply:
Huhu, good point. So, you could replace `LT.toStrict . Builder.toLazyText . build` with `pretty` (https://hackage.haskell.org/package/fmt-0.6.1.2/docs/Fmt.html#v:pretty) :+1:

Reply:
indeed! thanks